### DISCUSSION TITLE
Zigbee2MQTT + Cloudflare su rete mobile (SIM Card)

### DISCUSSION THREAD

**Discussion by vincenzocoletta** [indomus/forum#2962] [GENERALE - Discussioni generiche]
Ciao, a qualcuno risultano problemi relativi al tema in subject? Ho due installazioni identiche, entrambe raggiungibili da remoto tramite  tunneling Cloudflare e l'unica differenza tra le due che una è connessa tramite fibra mentre l'altra tramite Router Mobile (Fritz.box LTE con SIM VF). Su quest'ultimo proprio non riesco ad avviare Zigbee2MQTT... copio il report da registro:

```

[18:50:16] INFO: Preparing to start...
[18:50:16] INFO: Socat not enabled
[18:50:17] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:info  2024-03-08 18:50:19: Logging to console and directory: '/config/zigbee2mqtt/log/2024-03-08.18-50-19' filename: log.txt
Zigbee2MQTT:info  2024-03-08 18:50:19: Starting Zigbee2MQTT version 1.36.0 (commit #unknown)
Zigbee2MQTT:info  2024-03-08 18:50:19: Starting zigbee-herdsman (0.35.1)
Zigbee2MQTT:error 2024-03-08 18:50:39: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-03-08 18:50:39: Failed to start zigbee
Zigbee2MQTT:error 2024-03-08 18:50:39: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-03-08 18:50:39: Exiting...
Zigbee2MQTT:error 2024-03-08 18:50:39: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:62:27)
    at Controller.start (/app/lib/controller.ts:109:27)
    at start (/app/index.js:107:5)

```

 
<img width="883" alt="Screenshot 2024-03-08 alle 19 08 33" src="https://github.com/indomus/forum/assets/152210697/a4d8fc9c-e250-49e4-a378-dba8d0d87306">

**Comment by snakuzzo** [comment]
Scusa eh ...ma cosa potrebbe mai avere a che fare Cloudflare con zigbee2mqtt?
Lì l"errore è chiaro... zigbee2mqtt non comunica con l'hardware che usi come coordinator, quindi non potrà mai avviarsi e fare partire il servizio che espone la webgui, e di conseguenza non potrai mai aprire la webgui su HA

**Reply by vincenzocoletta** [reply]
Ti seguo ma non capisco cosa mi suggerisci... perdonami ma la mia conoscenza su alcuni temi è basic...

**Reply by vincenzocoletta** [reply]
Poi aggiungo che le due configurazioni sono perfettamente identiche, copiate e incollate (a meno dei parametri locali ovviamente...). Come mai una funziona (quella su fibra) e l'altra su mobile no?

**Reply by snakuzzo** [reply]
Zigbee2mqtt non comunica con la chiavetta per qualche motivo (su cui dovresti indagare tu)... non la vede... più semplice di così non so come dirtelo

```
Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
```

**Reply by vincenzocoletta** [reply]
Non capisco perchè ZHA (l'ho installato ora solo per conferma e poi disabilitato) la veda e Zigbee2MQTT no... vabbè, ci sbatterò un po la testa stasera... Intanto grazie! :)

**Reply by snakuzzo** [reply]
ma guarda che per farla vedere a zigbee2mqtt devi disabilitare ZHA. Non puoi (salvo magheggi inutili che creerebbero solo confusione) usare lo stesso hardware con due software diversi

**Reply by snakuzzo** [reply]
e...se non la vede, vuol dire che hai sbagliato la configurazione su zigbee2mqtt. Di che hardware stiamo parlando ? Che chiavetta hai ? Fai vedere la configurazione di zigbee2mqtt

**Reply by vincenzocoletta** [reply]
Pensa che come scrivevo già prima, forse non ci hai fatto caso, ZHA non l'avevo nemmeno installato. Come scrivevo l'ho caricato solo per vedere se la leggeva (e la vede) e poi l'ho disabilitato...
L'HW è: Sonoff Zigbee 3.0 Usb Dongle Plus E mentre la configurazione di Z2M è:

```

data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
mqtt:
  server: mqtt://core-mosquitto
  user: xxxxxxxx
  password: xxxxxxxx
serial:
  port: /dev/ttyACM1

```

Su porta di rete: 8484/tcp

Grazie per quello che potrai suggerirmi...

**Reply by vincenzocoletta** [reply]
Scusa, aggiungo HA OS su Raspberry pi5 8gb...

**Reply by snakuzzo** [reply]
La configurazione è sbagliata.
Se avessi letto con attenzione guida e/o documentazione (es. https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-home-assistant-hassio/ paragrafo dedicato al dongle-e) avresti visto che la configurazione dell'hw va fatta così...

Es. se la porta che hai indicato è corretta...

```yaml
serial:
  port: /dev/ttyACM1
  adapter: ezsp 
```

**Reply by vincenzocoletta** [reply]
Provo e ti dico. Però ti confermo una cosa di certo: la configurazione che ti ho copiato è identica al 100% a quella dell’altra istanza gemella (stesso HW e stessa configurazione) che però funziona… strano, no?

P.S. Porta HW verificata come primissima cosa e fatto copy&paste dalle impostazioni sistema hw per evitare typo

**Reply by snakuzzo** [reply]
Il dongle-e con zigbee2mqtt può funzionare solo come ti ho fatto vedere io, questo perché di base, se non lo metti, si aspetta hw con chip di Texas i strumenti, cosa che non ha il dongle-e.

**Reply by vincenzocoletta** [reply]
E tu sei numero 1 (ma questo te lo avevo già scritto :p), chapeau ancora una volta, funziona (aggiungendo “adapter: ezsp“). Però ti assicuro, i due HW sono assolutamente gemelli in tutto e per tutto e la configurazione di Z2M è la stessa al 100% (anche le user e pw le ho settate uguali) però… l’altro funziona. L’ho controllato, credimi, 1000 volte! Addirittura una ultima prima di scriverti adesso… Boh? Importante è che funzioni, grazie ancora!

**Reply by snakuzzo** [reply]
L'altro sarà un dongle-p altrimenti non funzionerebbe.

**Reply by vincenzocoletta** [reply]
Bah… comprati assieme su Amazon, ora ci guardo. Si riconosce in qualche modo in HA? Nelle impostazioni, hw hanno la stessa codifica…

**Reply by vincenzocoletta** [reply]
Confezioni con dettaglio modello allegate. Boh?

![IMG_3554](https://github.com/indomus/forum/assets/152210697/3960732a-8b04-4dcb-b38b-01dce7a0dfc3)

**Reply by vincenzocoletta** [reply]
Codifica HW identica su entrambe le installazioni (ne riporto una delle due):

![IMG_3557](https://github.com/indomus/forum/assets/152210697/218e7b26-2bfd-43dc-b012-504740aabd1e)

**Comment by vincenzocoletta** [comment]
Ciao, piccolo aggiornamento che potrebbe essere utile alla comunità: dopo aver sbattuto la testa sul motivo per cui i due dongle, pur essendo identici, uno funzionava con la il setting dell'adapter:

```

  adapter: ezsp 

```

e l'altro no ci ho rinunciato fino a quando stamane dopo un riavvio per mancanza tensione uno dei due smette di funzionare di nuovo. Gira e rigira credo di aver trovato l'arcano (o almeno adesso funzionano entrambi sia con che senza l'ezsp). Il problema invece sta nela porta deve che essere indicata in configurazione con il percorso intero:

```

serial:
  port: /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230829164715-if00

```

perché quello sintetico:

```

serial:
  port: /dev/ttyACM1

```

sembra essere dinamico nella parte numerica e quindi può cambiare (tipo a seguito di un reboot). Spero possa essere utile...

**Reply by snakuzzo** [reply]
Diciamo che non si tratta di percorso sintetico oppure no, ma di udev :) però diamola per buona.
Invece, non mi torna per niente il fatto che funzioni senza specificare `ezsp`

**Comment by vincenzocoletta** [comment]
Ma tu lo sai che io sono alle basi più assolute sui tecnicismi e udev non so nemmeno cosa sia… 🤣 Per quanto riguarda ezsp ti garantisco di ave provato più volte e che io lo metta o meno è completamente indifferente…
